src/clj/game/cards/agendas.clj

```diff
@@ -291,6 +291,23 @@
     :end-turn {:effect (effect (lose :runner :tag 2))
                :msg "make the Runner lose 2 tags"}}
 
+   "Broad Daylight"
+   {:prompt "Take 1 bad publicity?"
+    :choices ["Yes" "No"]
+    :async true
+    :effect (req (when (= target "Yes")
+                   (wait-for (gain-bad-publicity state :corp 1)
+                             (system-msg state :corp "used Broad Daylight to take 1 bad publicity")))
+                 (let [bp (get-in @state [:corp :bad-publicity] 0)]
```

Comment:
Is this going to include the BP from the above wait-for? Even if you have that trace asset and chose to not pay into the trace? Or Raymond Flint etc.

Reply:
Yes, I believe it should wait for everything that gets triggered from the bad-pub event to complete. Then the `state` should be up to date.

Reply:
@Saintis actually has the right idea. Anything that's not inside the `wait-for` will plow ahead if any `gain-bad-publicity` listeners are themselves async. I think there are no such interactions with gaining BP, but for safety the `let` should also be inside the wait-for.

Reply:
Does Broadcast Square affect this?

Reply:
Ah, right, I see. It's not actually blocking, we're just keeping track of what to follow up with when it's done.